velox/type/Type.h

```diff
@@ -2088,6 +2088,10 @@ struct CppToType<Timestamp> : public CppToTypeBase<TypeKind::TIMESTAMP> {};
 template <>
 struct CppToType<Date> : public CppToTypeBase<TypeKind::INTEGER> {};
 
+template <>
+struct CppToType<IntervalYearMonthType>
```

Comment:
Why is this needed?

Reply:
Without this change, it will fail to register a simple function with IntervalYearMonthType in its signature.

Reply:
That's strange. This PR doesn't register any function.

Reply:
This PR is extracted from https://github.com/facebookincubator/velox/pull/8887 which adds `make_ym_interval` function with return type of `IntervalYearMonthType`
I'll remove this change as it's irrelevant to this PR.